src/TLuaEngine.cpp

```diff
@@ -268,7 +268,7 @@ std::vector<std::string> TLuaEngine::StateThreadData::GetStateTableKeys(const st
 
     for (size_t i = 0; i < keys.size(); ++i) {
         auto obj = current.get<sol::object>(keys.at(i));
-        if (obj.get_type() == sol::type::nil) {
+        if (obj.get_type() == sol::type::none) {
```

Comment:
```suggestion
        if (obj.get_type() == sol::type::lua_nil) {
```

none is a different type (LUA_TNONE) signifying no type, whereas lua_nil is the proper way to do `nil`

Reply:
Alright, I will test that later when I'm back at home.

Reply:
changed to "lua_nil"  -> got successfull compiles on macos and my debian container. :)